baseapp/baseapp.go

```diff
@@ -1060,6 +1080,48 @@ func (app *BaseApp) runMsgs(ctx sdk.Context, msgs []sdk.Msg, msgsV2 []protov2.Me
 	}, nil
 }
 
+// simulateNestedMessages simulates a message nested messages.
+func (app *BaseApp) simulateNestedMessages(ctx sdk.Context, msg HasNestedMsgs) error {
+	msgs, err := msg.GetMsgs()
+	if err != nil {
+		return err
+	}
+
+	if err := validateBasicTxMsgs(app.msgServiceRouter, msgs); err != nil {
+		return err
+	}
+
+	msgsV2, err := app.msgsV1ToMsgsV2(msgs)
+	if err != nil {
+		return err
+	}
+
+	_, err = app.runMsgs(ctx, msgs, msgsV2, execModeSimulate)
+	return err
+}
+
+// msgsV1ToMsgsV2 transforms v1 messages into v2.
+func (app *BaseApp) msgsV1ToMsgsV2(msgs []sdk.Msg) ([]protov2.Message, error) {
+	msgsV2 := make([]protov2.Message, len(msgs))
+	for i, msg := range msgs {
+		gogoAny, err := codectypes.NewAnyWithValue(msg)
+		if err != nil {
+			return nil, err
+		}
+		anyMsg := &anypb.Any{
+			TypeUrl: gogoAny.TypeUrl,
+			Value:   gogoAny.Value,
+		}
+		msgV2, err := anyutil.Unpack(anyMsg, app.cdc.InterfaceRegistry().SigningContext().FileResolver(), app.cdc.InterfaceRegistry().SigningContext().TypeResolver())
+		if err != nil {
+			return nil, err
+		}
+		msgsV2[i] = msgV2
+	}
```

Comment:
Can we use https://pkg.go.dev/google.golang.org/protobuf@v1.34.1/protoadapt#MessageV2Of instead of packing/unpacking?

Reply:
👏👏👏